cmd/clair-load-test/reports.go

```diff
@@ -170,28 +181,50 @@ func reportAction(c *cli.Context) error {
 		return fmt.Errorf("Invalid layer value. Must be one among: %v", validLayers)
 	}
 	if conf.TestRepoPrefix != "" {
-		conf.Containers = getContainersList(ctx, conf.TestRepoPrefix, conf.HitSize, conf.Layers, validLayers)
+		warmUpContainers = getContainersList(ctx, conf.TestRepoPrefix, 1, conf.WarmupSize, conf.Layers, validLayers)
+		conf.Containers = getContainersList(ctx, conf.TestRepoPrefix, conf.WarmupSize+1, conf.HitSize, conf.Layers, validLayers)
 	}
 	if len(conf.Containers) > conf.HitSize {
 		conf.Containers = conf.Containers[:conf.HitSize]
 	}
-	listOfManifests, listOfManifestHashes := manifests.GetManifest(ctx, conf.Containers, conf.Concurrency)
 	jwt_token, err := CreateToken(conf.PSK)
 	if err != nil {
 		zlog.Debug(ctx).Str("PSK", conf.PSK).Msg("creating token")
 		return fmt.Errorf("could not create token: %w", err)
 	}
-	err = orchestrateWorkload(ctx, listOfManifests, listOfManifestHashes, jwt_token, conf)
+
+	zlog.Debug(ctx).Msg("Fetching manifests for an actual workload")
+	listOfManifests, listOfManifestHashes := manifests.GetManifest(ctx, conf.Containers, conf.Concurrency)
+	if len(warmUpContainers) > 0 {
+		zlog.Debug(ctx).Msg("Fetching manifests for warmup")
+		warmpUpManifests, warmUpManifestHashes := manifests.GetManifest(ctx, warmUpContainers, conf.Concurrency)
+		zlog.Info(ctx).Msg("Downloading manifests completed\n")
+		if len(warmpUpManifests) > 0 {
+			zlog.Info(ctx).Msg("🧨 Running a warmup before actual workload")
+			warmUpConf := deepcopy.Copy(conf).(*TestConfig)
```

Comment:
Why does this need a deep copy? That seems to indicate this isn't actually a configuration and is modified.

Reply:
In the warmup step, we did not want to index the results to ES, so did a clone of existing config and setup some dummy params so that we can just use it for warmup and not store any of those results.